Test failures with new 0.26.2 release

After https://github.com/libgit2/pygit2/pull/744 pygit2 self tests are randomly failing with what looks like memory corruption:

```
======================================================================
FAIL: test_patch_create_from_blob_buffer (test.test_patch.PatchTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/pygit2-0.26.2/test/test_patch.py", line 110, in test_patch_create_from_blob_buffer
    self.assertEqual(patch.patch, BLOB_PATCH)
AssertionError: u'diff --git a/a/file b/b/file\nindex a520c24..d675fa4 100644\n--- a/a/file\n+++ [truncated]... != u'diff --git a/a/file b/b/file\nindex a520c24..d675fa4 100644\n--- a/a/file\n+++ [truncated]...
  diff --git a/a/file b/b/file
  index a520c24..d675fa4 100644
  --- a/a/file
  +++ b/b/file
  @@ -1,3 +1 @@
- -+ -hello world
+ -hola mundo
+ -bonjour le monde
+ +foo bar
======================================================================
FAIL: test_patch_create_from_blob_buffer_delete (test.test_patch.PatchTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/pygit2-0.26.2/test/test_patch.py", line 132, in test_patch_create_from_blob_buffer_delete
    self.assertEqual(patch.patch, BLOB_PATCH_DELETED)
AssertionError: u'diff --git a/a/file b/b/file\ndeleted file mode 100644\nindex a520c24..0000000 [truncated]... != u'diff --git a/a/file b/b/file\ndeleted file mode 100644\nindex a520c24..0000000 [truncated]...
  diff --git a/a/file b/b/file
  deleted file mode 100644
  index a520c24..0000000
  --- a/a/file
  +++ /dev/null
  @@ -1,3 +0,0 @@
- -+ -hello world
+ -hola mundo
+ -bonjour le monde
======================================================================
FAIL: test_patch_create_from_blobs (test.test_patch.PatchTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/pygit2-0.26.2/test/test_patch.py", line 99, in test_patch_create_from_blobs
    self.assertEqual(patch.patch, BLOB_PATCH)
AssertionError: u'diff --git a/a/file b/b/file\nindex a520c24..d675fa4 100644\n--- a/a/file\n+++ [truncated]... != u'diff --git a/a/file b/b/file\nindex a520c24..d675fa4 100644\n--- a/a/file\n+++ [truncated]...
  diff --git a/a/file b/b/file
  index a520c24..d675fa4 100644
  --- a/a/file
  +++ b/b/file
  @@ -1,3 +1 @@
- -0\ufffd- @+ -hello world
+ -hola mundo
+ -bonjour le monde
+ +foo bar
----------------------------------------------------------------------
```

--- a/src/patch.c
+++ b/src/patch.c
@@ -108,7 +108,7 @@ PyDoc_STRVAR(Patch_create_from__doc__,
     "Create a patch from blobs, buffers, or a blob and a buffer");
 
 static PyObject *
-Patch_create_from(PyObject *self, PyObject *args)
+Patch_create_from(PyObject *self, PyObject *args, PyObject *kwds)
 {
   /* A generic wrapper around
    * git_patch_from_blob_and_buffer
@@ -124,8 +124,11 @@ Patch_create_from(PyObject *self, PyObject *args)
   Py_ssize_t oldbuflen, newbuflen;
   int err;
 
-  if (!PyArg_ParseTuple(args, "OzOz|I", &oldobj, &old_as_path, &newobj,
-                        &new_as_path, &opts.flags))
+  char *keywords[] = {"old", "new", "flag", "old_as_path", "new_as_path", NULL};
+
+  if (!PyArg_ParseTupleAndKeywords(args, kwds, "OO|Izz", keywords,
+                                   &oldobj, &newobj, &opts.flags,
+                                   &old_as_path, &new_as_path))
     return NULL;
 
   if (oldobj != Py_None && PyObject_TypeCheck(oldobj, &BlobType))
@@ -201,7 +204,8 @@ cleanup:
 }
 
 PyMethodDef Patch_methods[] = {
-    {"create_from", (PyCFunction) Patch_create_from, METH_VARARGS | METH_STATIC, Patch_create_from__doc__},
+    {"create_from", (PyCFunction) Patch_create_from,
+      METH_KEYWORDS | METH_VARARGS | METH_STATIC, Patch_create_from__doc__},
     {NULL}
 };
 

--- a/test/test_patch.py
+++ b/test/test_patch.py
@@ -31,15 +31,26 @@ from __future__ import unicode_literals
 import pygit2
 from . import utils
 
-BLOB_SHA = 'a520c24d85fbfc815d385957eed41406ca5a860b'
-BLOB_OLD_CONTENT = """hello world
+BLOB_OLD_SHA = 'a520c24d85fbfc815d385957eed41406ca5a860b'
+BLOB_NEW_SHA = '3b18e512dba79e4c8300dd08aeb37f8e728b8dad'
+BLOB_OLD_CONTENT = b"""hello world
 hola mundo
 bonjour le monde
-""".encode()
+"""
 BLOB_NEW_CONTENT = b'foo bar\n'
 
 BLOB_OLD_PATH = 'a/file'
 BLOB_NEW_PATH = 'b/file'
+
+BLOB_PATCH2 = """diff --git a/a/file b/b/file
+index a520c24..3b18e51 100644
+--- a/a/file
++++ b/b/file
+@@ -1,3 +1 @@
+ hello world
+-hola mundo
+-bonjour le monde
+"""
 
 BLOB_PATCH = """diff --git a/a/file b/b/file
 index a520c24..d675fa4 100644
@@ -78,33 +89,33 @@ class PatchTest(utils.RepoTestCase):
     def test_patch_create_from_buffers(self):
         patch = pygit2.Patch.create_from(
             BLOB_OLD_CONTENT,
-            BLOB_OLD_PATH,
             BLOB_NEW_CONTENT,
-            BLOB_NEW_PATH,
+            old_as_path=BLOB_OLD_PATH,
+            new_as_path=BLOB_NEW_PATH,
         )
 
         self.assertEqual(patch.patch, BLOB_PATCH)
 
     def test_patch_create_from_blobs(self):
-        old_blob = self.repo.create_blob(BLOB_OLD_CONTENT)
-        new_blob = self.repo.create_blob(BLOB_NEW_CONTENT)
+        old_blob = self.repo[BLOB_OLD_SHA]
+        new_blob = self.repo[BLOB_NEW_SHA]
 
         patch = pygit2.Patch.create_from(
-            self.repo[old_blob],
-            BLOB_OLD_PATH,
-            self.repo[new_blob],
-            BLOB_NEW_PATH,
+            old_blob,
+            new_blob,
+            old_as_path=BLOB_OLD_PATH,
+            new_as_path=BLOB_NEW_PATH,
         )
 
-        self.assertEqual(patch.patch, BLOB_PATCH)
+        self.assertEqual(patch.patch, BLOB_PATCH2)
 
     def test_patch_create_from_blob_buffer(self):
-        old_blob = self.repo.create_blob(BLOB_OLD_CONTENT)
+        old_blob = self.repo[BLOB_OLD_SHA]
         patch = pygit2.Patch.create_from(
-            self.repo[old_blob],
-            BLOB_OLD_PATH,
+            old_blob,
             BLOB_NEW_CONTENT,
-            BLOB_NEW_PATH,
+            old_as_path=BLOB_OLD_PATH,
+            new_as_path=BLOB_NEW_PATH,
         )
 
         self.assertEqual(patch.patch, BLOB_PATCH)
@@ -112,21 +123,21 @@ class PatchTest(utils.RepoTestCase):
     def test_patch_create_from_blob_buffer_add(self):
         patch = pygit2.Patch.create_from(
             None,
-            BLOB_OLD_PATH,
             BLOB_NEW_CONTENT,
-            BLOB_NEW_PATH,
+            old_as_path=BLOB_OLD_PATH,
+            new_as_path=BLOB_NEW_PATH,
         )
 
         self.assertEqual(patch.patch, BLOB_PATCH_ADDED)
 
     def test_patch_create_from_blob_buffer_delete(self):
-        old_blob = self.repo.create_blob(BLOB_OLD_CONTENT)
+        old_blob = self.repo[BLOB_OLD_SHA]
 
         patch = pygit2.Patch.create_from(
-            self.repo[old_blob],
-            BLOB_OLD_PATH,
+            old_blob,
             None,
-            BLOB_NEW_PATH,
+            old_as_path=BLOB_OLD_PATH,
+            new_as_path=BLOB_NEW_PATH,
         )
 
         self.assertEqual(patch.patch, BLOB_PATCH_DELETED)
@@ -135,16 +146,12 @@ class PatchTest(utils.RepoTestCase):
         with self.assertRaises(TypeError):
             pygit2.Patch.create_from(
                 self.repo,
-                BLOB_OLD_PATH,
                 BLOB_NEW_CONTENT,
-                BLOB_NEW_PATH,
             )
 
     def test_patch_create_from_bad_new_type_arg(self):
         with self.assertRaises(TypeError):
             pygit2.Patch.create_from(
                 None,
-                BLOB_OLD_PATH,
                 self.repo,
-                BLOB_NEW_PATH,
             )

